projects/adapter/src/dbt/adapters/fal_experimental/utils/environments.py

```diff
@@ -39,42 +37,41 @@ def run(self, executable: BasicCallable, *args, **kwargs) -> Any:
 
 
 @dataclass
-class KoldstartDummyServer(IsolateServer):
-    machine_type: str
-    target_env_type: str
-    creds: CloudKeyCredentials | None = None
+class EnvironmentDefinition():
+    host: Host
+    kind: str
+    config: dict[Any, Any]
+    machine_type: str = "S"
 
 
 def fetch_environment(
     project_root: str,
     environment_name: str,
     machine_type: str = "S",
     credentials: Optional[Any] = None,
-) -> Tuple[BaseEnvironment, bool]:
+) -> Tuple[EnvironmentDefinition, bool]:
     """Fetch the environment with the given name from the project's
     fal_project.yml file."""
     # Local is a special environment where it doesn't need to be defined
     # since it will mirror user's execution context directly.
     if environment_name == "local":
         if credentials.key_secret and credentials.key_id:
-            # "local" environment in this case is fal cloud
-            # TODO: add docs link
             logger.warning(
                 "`local` environments will be executed on fal cloud."
                 + "If you don't want to use fal cloud, you can change your "
                 + "profile target to one where fal doesn't have credentials"
             )
```

Comment:
what do you think of actually deciding this not on credentials but on `host` being there?

Reply:
since you recently added the `host: cloud`, I think it would make sense